src/domains/User/context/UserProvider.js

```diff
@@ -90,20 +91,26 @@ const UserProvider = ({ children }) => {
       !userPreloading && user && !value?.email && !bookingLoading
     /* If there is no user data in the database, save the user data to the database. */
     if (isNoUserDataInDB) {
-      const [firstName, lastName] = user?.displayName?.split(' ') || [
-        null,
-        null
-      ]
-      saveUserToDB({
-        _id: user.uid,
-        email: user.email,
-        avatarUrl: user.photoURL,
-        agreement: true,
-        firstName: firstName,
-        lastName: lastName,
-        phoneNumber: user.phoneNumber || null,
-        gdpr,
-        onError: handleError
+      // Check above sometimes not working correctly, so
+      // additional check for user doc existing was added, to prevent overwriting user data
+      getDocument(COLLECTIONS.USERS, user.uid).then((doc) => {
```

Comment:
Where is the .catch section?
What will be if there is an error during getDocument operation for whatever reason?